data/src/main/java/com/example/data/repository/GitHubRepositoryImpl.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.data.repository
+
+import ResultWrapper
+import com.example.data.remote.datasource.GitHubDataSource
+import com.example.data.remote.response.search.Item
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flatMapMerge
+import kotlinx.coroutines.flow.flow
+import model.SearchModel
+import repository.GitHubRepository
+import javax.inject.Inject
+
+class GitHubRepositoryImpl @Inject constructor(
+    private val gitHubDataSource: GitHubDataSource
+) : GitHubRepository {
+
+    @OptIn(FlowPreview::class)
+    override fun getSearchModel(query: String): Flow<ResultWrapper<Any, Any>> = channelFlow {
+        gitHubDataSource.searchUsers(query).catch { e ->
+            send(ResultWrapper.Fail("SearchError[searchUsers]: $e"))
+        }.flatMapMerge { searchUserResponse ->
+            combineSearchList(searchUserResponse.items ?: emptyList())
+        }.collect {
+            send(it)
+        }
+    }
+
+    private fun combineSearchList(list: List<Item>) = channelFlow {
```

Comment:
저는 combine으로 호출해본 적은 처음이어서 에러처리를 어떻게 가져갈지 고민이었는데 새로운 방법인 것 같아서 배워갑니다! channelFow도...👍🏻

Reply:
사실 channelFlow를 안 쓸 수 있더라구용. 그냥 flow 빌더로 열었을 때 디스패처를 명시해주지 않았더니 아래와 같은 예외 발생 시오류가 났어요
```
Emission from another coroutine is detected.
Child of ProducerCoroutine{Active}@b38275c, expected child of StandaloneCoroutine{Active}@efa65.
FlowCollector is not thread-safe and concurrent emissions are prohibited.
To mitigate this restriction please use 'channelFlow' builder instead of 'flow'
```
해결법은 찾은 게 2개였는데 channelFlow빌더를 사용하거나 flowOn(Dispatchers.IO)를 통해 디스패처를 명시해 주는거 였는데 아직 완벽하게 공부가 되진 않았습니다.